contracts/migrator/README.md

```diff
@@ -0,0 +1,193 @@
+- **WHEN** `repayAmount == type(uint256).max)`:
+  - **BIND READ** `repayAmountActual = cToken.borrowBalanceCurrent(user)`
+- **ELSE**
+  - **BIND** `repayAmountActual = repayAmount`
+- **BIND** `borrowAmountTotal = repayAmountActual + FullMath.mulDivRoundingUp(repayAmountActual, uniswapLiquidityPoolFee, 1e6)` # TODO: FullMath
+- **BIND** `data = abi.encode(MigrationCallbackData{user, repayAmountActual, repayBorrowBehalf, collateral})`
+- **CALL** `uniswapLiquidityPool.flash(address(this), uniswapLiquidityPoolToken0 ? repayAmount : 0, uniswapLiquidityPoolToken0 ? 0 : repayAmount, data)`
+- **STORE** `inMigration -= 1`
+
+Note: for fee calculation see [UniswapV3Pool](https://github.com/Uniswap/v3-core/blob/main/contracts/UniswapV3Pool.sol#L800).
+
+### Uniswap Liquidity Pool Callback Function
+
+This function handles a callback from the Uniswap Liquidity Pool after it has sent this contract the requested tokens. We are responsible for repaying those tokens, with a fee, before we return from this function call.
+
+#### Pre-conditions
+
+This function may only be called during a migration command. We ensure this by making sure this function, itself, is the caller of the `swap` command. We check that the call originates from the expected Uniswap pool, and we check that we are actively processing a migration. This combination of events should ensure that no external party can trigger this code, though it's not clear it would be dangerous even if such a party did.
+
+#### Inputs
+
+ - `uint fee0`: The fee for borrowing token0 from pool. Ignored.
+ - `uint fee1`: The fee for borrowing token1 from pool. Ignored.
+ - `calldata data`: The data encoded above, which is the ABI-encoding of `MigrationCallbackData`.
+
+#### Bindings
+
+ * `repayAmountActual`: The repay amount, after accounting for max.
+
+#### Function Spec
+
+`function uniswapV3FlashCallback(uint256 fee0, uint256 fee1, bytes calldata data)`
+
+  - **REQUIRE** `inMigration == 1`
+  - **REQUIRE** `msg.sender == uniswapLiquidityPool`
+  - **REQUIRE** `sender == address(this)`
+  - **BIND** `MigrationCallbackData{user, repayAmountActual, borrowAmountTotal, collateral} = abi.decode(data, (MigrationCallbackData))`
+  - **CALL** `borrowCToken.repayBorrowBehalf(user, repayAmountActual)`
+  - **FOREACH** `(cToken, amount)` in `collateral`:
+    - **CALL** `cToken.transferFrom(user, amount == type(uint256).max ? cToken.balanceOf(user) : amount)`
+    - **CALL** `cToken.redeem(cToken.balanceOf(address(this)))`
+    - **CALL** `comet.supplyCollateral(address(this), user, cToken.underlying(), cToken.underlying().balanceOf(address(this)))`
+  - **CALL** `comet.withdrawBase(user, address(this), borrowAmountTotal)`
+  - **CALL** `pay(borrowToken, address(this), msg.sender, borrowAmountTotal)`
+
+### Sweep Function
+
+Sends any tokens in this contract to the sweepee address. This contract should never hold tokens, so this is just to fix any anomalistic situations where tokens end up locked in the contract.
```

Comment:
Another way to do this is to have an `admin` that can sweep tokens to any recipient. That way, you wouldn't have to hardcode the `sweepee` and could set the `admin` as the `Timelock`. That's how we set it up in the `Bulker`.